src/Workspaces/Core/Portable/Rename/ConflictEngine/RelatedLocation.cs

```diff
@@ -12,62 +11,23 @@ namespace Microsoft.CodeAnalysis.Rename.ConflictEngine
     /// </summary>
+    /// <param name="ConflictCheckSpan">
+    /// The Span of the original identifier if it was in source, otherwise the span to check for implicit
+    /// references.
+    /// </param>
+    /// <param name="ComplexifiedTargetSpan">
+    /// If there was a conflict at ConflictCheckSpan during rename, then the next phase in rename uses
+    /// ComplexifiedTargetSpan span to be expanded to resolve the conflict.
+    /// </param>
     [DataContract]
-    internal readonly struct RelatedLocation(TextSpan conflictCheckSpan, DocumentId documentId, RelatedLocationType type, bool isReference = false, TextSpan complexifiedTargetSpan = default) : IEquatable<RelatedLocation>
+    internal readonly record struct RelatedLocation(
+        [field: DataMember(Order = 0)] TextSpan ConflictCheckSpan,
+        [field: DataMember(Order = 1)] DocumentId DocumentId,
+        [field: DataMember(Order = 2)] RelatedLocationType Type,
+        [field: DataMember(Order = 3)] bool IsReference = false,
+        [field: DataMember(Order = 4)] TextSpan ComplexifiedTargetSpan = default)
     {
-        /// <summary>
-        /// The Span of the original identifier if it was in source, otherwise the span to check for implicit
-        /// references.
-        /// </summary>
-        [DataMember(Order = 0)]
-        public readonly TextSpan ConflictCheckSpan = conflictCheckSpan;
-
-        [DataMember(Order = 1)]
-        public readonly DocumentId DocumentId = documentId;
-
-        [DataMember(Order = 2)]
-        public readonly RelatedLocationType Type = type;
-
-        [DataMember(Order = 3)]
-        public readonly bool IsReference = isReference;
-
-        /// <summary>
-        /// If there was a conflict at ConflictCheckSpan during rename, then the next phase in rename uses
-        /// ComplexifiedTargetSpan span to be expanded to resolve the conflict.
-        /// </summary>
-        [DataMember(Order = 4)]
-        public readonly TextSpan ComplexifiedTargetSpan = complexifiedTargetSpan;
-
         public RelatedLocation WithType(RelatedLocationType type)
             => new(ConflictCheckSpan, DocumentId, type, IsReference, ComplexifiedTargetSpan);
-
-        public override bool Equals(object? obj)
-            => obj is RelatedLocation location && Equals(location);
-
-        public bool Equals(RelatedLocation other)
-        {
-            return ConflictCheckSpan.Equals(other.ConflictCheckSpan) &&
-                   Type == other.Type &&
-                   IsReference == other.IsReference &&
-                   EqualityComparer<DocumentId>.Default.Equals(DocumentId, other.DocumentId) &&
-                   ComplexifiedTargetSpan.Equals(other.ComplexifiedTargetSpan);
-        }
-
-        public override int GetHashCode()
-        {
-            var hashCode = 928418920;
-            hashCode = hashCode * -1521134295 + ConflictCheckSpan.GetHashCode();
-            hashCode = hashCode * -1521134295 + ((int)Type).GetHashCode();
```

Comment:
⚠️ Removing this might be forcing an allocation to occur (enum boxing on .NET Framework)

Reply:
this is only during an explicit rename.  i do not envision this mattering in practice.

Reply:
➡️ I reviewed the history, and it appears this change was the result of Banned API Analyzers flagging all uses of Enum.GetHashCode (including ones that were not problematic). Since this method wasn't marked with `[PerformanceSensitive]` (while the other ones that hit the boxing problem were), it seems safe to assume this change is not problematic even if an allocation results from it.